Python/omevv/v1/scripts/discoverServerDevice.py

```diff
@@ -0,0 +1,110 @@
+
+        hostDiscoveryGroupPayload = []
+        hostDiscoveryGroupPayload.append(
+            {
+                "consoleEntityIDs": [self.console_entity_id],
+                "userName": self.device_username,
+                "passWord": self.device_password,
+                "useGlobalCredentials": self.use_global_creds            
+            }
+        )
+
+        if self.console_entity_id and self.device_username and self.device_password:
+            self.payload["hostDiscoveryGroups"] = hostDiscoveryGroupPayload;
+
+    def run_discovery(self):
+        global retry
+        headers = self.headers
+        url = 'https://%s/omevv/GatewayService/v1/Consoles/%s/Hosts/Discover'%(self.omeIp, self.uuid)
+        try:
+            response = requests.post(url, data=json.dumps(self.payload), headers=headers, verify=False);
+            data = response.json();
+            status_code = response.status_code
+            if status_code == 202:
+              return "Discovery job is created successfully with id "+ str(data)
+            elif status_code == 400 or status_code == 500 or status_code == 404:
+                return "Error occured while creating discovery job : "+ str(data)
+            else:
+                raise Exception("Error occured while creating discovery job ",data);
+        except Exception as e:
+            print("Exception occured while creating discovery job ",e," retrying ..");
+            if retry > 0:
+                retry = retry - 1;
+                time.sleep(5);
+                self.run_discovery();
+
+            else:
+                print("Failed after 3 retries,exiting");
+                sys.exit();
+
+if __name__ == "__main__":
+    PARSER = argparse.ArgumentParser(description=__doc__,
+                                     formatter_class=argparse.RawTextHelpFormatter)
+    PARSER.add_argument("--ip", "-i", required=True, help="OME Appliance IP")
+    PARSER.add_argument("--vcusername", "-u", required=True,
+                        help="username of vcenter")
+    PARSER.add_argument("--vcpassword", "-p", required=True,
+                        help="password of vcenter")
+    PARSER.add_argument("--vcUUID", "-d", required=True, default=None,
+                        help="UUID of the relevant vCenter")
+    PARSER.add_argument("--jobname", "-g", required=True, default=None, help="job name")
+    PARSER.add_argument("--jobdescription", required=False, default=None, help="job description")
+    PARSER.add_argument("--console_entity_id", required=True, default=None, help="console entity id of the server device")
+    PARSER.add_argument("--device_username", required=True, default=None, help="username of device")
+    PARSER.add_argument("--device_password", required=True, default=None, help="password of device")
+    PARSER.add_argument("--use_global_creds", required=True, default=None, help="indicate whether to use global credentials")
```

Comment:
Are we telling user what is the global creds? If global creds are true, why do we need device username and password?